dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/validation/hooks/PreCheckOwnershipValidationHook.java

```diff
@@ -257,19 +257,23 @@ public void validateEvent( ValidationErrorReporter reporter, Event event )
         // Check acting user is allowed to change existing/write event
         if ( strategy.isUpdateOrDelete() )
         {
-            validateUpdateAndDeleteEvent( reporter, event, context.getProgramStageInstance( event.getEvent() ),
+            ProgramStageInstance programStageInstance = context.getProgramStageInstance( event.getEvent() );
+            TrackedEntityInstance entityInstance = programStageInstance.getProgramInstance().getEntityInstance();
+            validateUpdateAndDeleteEvent( reporter, event, programStageInstance,
```

Comment:
Maybe out of the scope of this PR but can we make signature of this method more readable like passing programStageInstance once?

Reply:
Done 👍